cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/relational/ExpressionFactory.java

```diff
@@ -49,15 +49,15 @@
    * Provides fully qualified name to be used in expressions for a given dataset.
    * Available with {@link CoreExpressionCapabilities#CAN_GET_QUALIFIED_DATASET_NAME}
    */
-  default String getQualifiedDataSetName(Relation dataSet) {
+  default ExtractableExpression<T> getQualifiedDataSetName(Relation dataSet) {
     throw new UnsupportedOperationException();
```

Comment:
I would be good to have it return an invalid expression, but it can be done in a follow-up PR

Reply:
Note: `setDataSetAlias` below may also return and invalid Relation by default